config/utilities/uenv.py

```diff
@@ -0,0 +1,107 @@
+import os
+import pathlib
+import yaml
+
+import reframe.utility.osext as osext
+from reframe.core.exceptions import ConfigError
+from packaging.version import Version
+
+
+_UENV_MOUNT_DEFAULT = '/user-environment'
+_UENV_CLI = 'uenv'
+_UENV_DELIMITER = ','
+_UENV_MOUNT_DELIMITER = '@'
+_RFM_META = pathlib.Path('extra') / 'reframe.yaml'
+
+
+def _get_uenvs():
+    uenv = os.environ.get('UENV', None)
+    if uenv is None:
+        return uenv 
+
+    uenv_environments = []
+    uenv_list = uenv.split(_UENV_DELIMITER)
+    uenv_version = osext.run_command(
+        f'{_UENV_CLI} --version', shell=True
+    ).stdout.strip()
+
+    for uenv in uenv_list:
+        uenv_name, *image_mount = uenv.split(_UENV_MOUNT_DELIMITER)
+        if len(image_mount) > 0:
+            image_mount = image_mount[0]
+        else:
+            image_mount = _UENV_MOUNT_DEFAULT 
+
+        inspect_cmd = f'{_UENV_CLI} image inspect {uenv_name} --format'
+
+        image_path = osext.run_command(f"{inspect_cmd} '{{sqfs}}'", shell=True
+            ).stdout.strip()
+        target_system = osext.run_command(f"{inspect_cmd} '{{system}}'",
+            shell=True).stdout.strip()
+
+        image_path = pathlib.Path(image_path)
+
+        #FIXME temporary workaround for older uenv versions
+        if Version(uenv_version) >= Version('5.1.0-dev'):
+            meta_path = osext.run_command(
+                f"{inspect_cmd} '{{meta}}'", shell=True
+            ).stdout.strip()
+            rfm_meta = pathlib.Path(meta_path) / _RFM_META
+        else:
+            rfm_meta = image_path.parent / 'store.yaml'
+
+        try:
+            with open(rfm_meta) as image_envs:
+                image_environments = yaml.load(
+                    image_envs.read(), Loader=yaml.BaseLoader)
```

Comment:
```suggestion
                    image_envs.read(), Loader=yaml.BaseLoader)
                    print(f"# --- loading the metadata from '{rfm_meta}'")
```

Reply:
I don't want to add this because it messes up the `--show-config` if I want to parse with `jq`